core/lib/object_store/src/gcs_unauthenticated.rs

```diff
@@ -0,0 +1,99 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use http::StatusCode;
+use reqwest::Error;
+
+use crate::{raw::BoxedError, Bucket, ObjectStore, ObjectStoreError};
+
+#[derive(Debug)]
+pub struct UnauthenticatedGoogleCloudStorage {
+    bucket_prefix: String,
+    max_retries: u16,
+}
+
+impl UnauthenticatedGoogleCloudStorage {
+    pub fn new(bucket_prefix: String, max_retries: u16) -> UnauthenticatedGoogleCloudStorage {
+        Self {
+            bucket_prefix,
+            max_retries,
+        }
+    }
+}
+
+#[async_trait]
+impl ObjectStore for UnauthenticatedGoogleCloudStorage {
+    async fn get_raw(&self, bucket: Bucket, key: &str) -> Result<Vec<u8>, ObjectStoreError> {
+        const DOWNLOAD_TIMEOUT: Duration = Duration::from_secs(600);
+        let client = reqwest::Client::builder()
+            .timeout(DOWNLOAD_TIMEOUT)
+            .build()
+            .map_err(|error| ObjectStoreError::Other(error.into()))?;
+
+        let mut last_error: Option<Error> = None;
+        let retry_count = self.max_retries;
+        for retry_number in 0..retry_count {
+            let url = format!("{}/{key}", self.storage_prefix_raw(bucket));
+            let response = client.get(url.to_string()).send().await;
+
+            match response {
+                Ok(response) => {
+                    let response_code = response.status();
+                    match response_code {
+                        StatusCode::NOT_FOUND => {
+                            let error_message = format!(
+                                "missing key: {key} in bucket {bucket} (got 404 from {url})"
+                            );
+                            return Err(ObjectStoreError::KeyNotFound(error_message.into()));
+                        }
+                        StatusCode::OK => {}
+                        _ => {
+                            let error_message = format!("unexpected error when fetching {key} from bucket {bucket}, received code {response_code}");
+                            return Err(ObjectStoreError::Other(error_message.into()));
+                        }
+                    }
+                    let bytes = response.bytes().await.map(|bytes| bytes.to_vec());
+                    match bytes {
+                        Ok(bytes) => return Ok(bytes),
+                        Err(error) => {
+                            last_error = Some(error);
+                        }
+                    }
+                }
+                Err(error) => {
+                    last_error = Some(error);
+                }
+            }
+
+            tracing::warn!(
+                "Failed to download {url} (attempt {}/{retry_count}). Backing off for 5 second",
+                retry_number + 1
+            );
+            tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
Nit: You can extract the timeout into a constant and use it in the log message above.